File: scalad.core/src/main/java/scalad/CriteriaWrapper.java
package scalad;

import javax.persistence.EntityManager;
import javax.persistence.TypedQuery;
import javax.persistence.criteria.CriteriaBuilder;
import javax.persistence.criteria.CriteriaQuery;
import javax.persistence.criteria.Predicate;
import javax.persistence.criteria.Root;

/**
 * @author janmachacek
 */
class CriteriaWrapper {

	static <T> TypedQuery<T> getQuery(Query queryBuilder, EntityManager entityManager, Class<T> entityType) {
		final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
		final CriteriaQuery<T> criteriaQuery = cb.createQuery(entityType);
		final Root<T> root = criteriaQuery.from(entityType);

		final Predicate predicate = cb.like(root.<String>get(queryBuilder.property()), queryBuilder.value().get().toString());

		criteriaQuery.where(predicate);

		return entityManager.createQuery(criteriaQuery);
	}
}

File: scalad.core/src/main/scala/scalad/CriteriaWrapper.scala
package scalad

import javax.persistence.EntityManager;


/**
 * @author janmachacek
 */
object CriteriaWrapper {

  def getQuery[T](query: Query, entityManager: EntityManager, entityType: Class[T]) = {
    val cb = entityManager.getCriteriaBuilder
    val criteriaQuery = cb.createQuery(entityType)
    val root = criteriaQuery.from(entityType)

    //val predicate = cb.like(root.get[String](query.property), query.value.get.toString)
    //criteriaQuery.where(Array(predicate):_*)

    entityManager.createQuery(criteriaQuery)
  }

}
